src/app/fb/db.cljs

```diff
@@ -1,11 +1,40 @@
 (ns app.fb.db
-  (:require ["firebase/app" :refer [database]]
+  (:require [re-frame.core :refer [reg-fx dispatch]]
+            [clojure.walk :refer [keywordize-keys]]
+            ["firebase/app" :as firebase]
             [clojure.string :as str]))
 
-(defn db-ref
-  [path]
-  (.ref (database) (str/join "/" path)))
+(defn materialize-col-snap
+  [col-snap]
+  (->> (.-docs col-snap)
+       (map #(.data %))
+       (map #(js->clj %))
+       (map keywordize-keys)
+       vec))
+
+(reg-fx
+  :persist
+  (fn [{:keys [path attrs]}]
+    (let [firestore (.firestore firebase)
+          doc-ref (.doc firestore path)]
+      (.set doc-ref (clj->js attrs) #js {:merge true}))))
+
+(reg-fx
+  :persist-delete
```

Comment:
hmm, maybe `persist` was not best idea, and it just would be better to name it `:firestore/save`, `:firestore/delete`, `:firestore/get-doc`, `:firestore/get-col` etc? no hidden intent, same naming as in firestore itself etc

Reply:
I like it very much :)